### PR TITLE
support London RPC API

### PR DESCRIPTION
**Pull request by lcmmhcc**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: app/rpc/types/types.go [hunk 1]
```diff
@@ -159,3 +160,22 @@ type EthHeaderWithBlockHash struct {
...
+	GasUsedRatio []float64        `json:"gasUsedRatio"`
+}
+
+// SignTransactionResult represents a RLP encoded signed transaction.
+type SignTransactionResult struct {
+	Raw hexutil.Bytes `json:"raw"`
+	Tx  *Transaction  `json:"tx"`
+}
+
+type OneFeeHistory struct {
```
Comment (ilovers) [inline app/rpc/types/types.go]: There is no need to add this struct, as no reference to this struct.

File: app/rpc/backend/backend.go [hunk 2]
```diff
@@ -52,6 +54,7 @@ type Backend interface {
 
 	// Used by eip-1898
 	ConvertToBlockNumber(rpctypes.BlockNumberOrHash) (rpctypes.BlockNumber, error)
+	FeeHistory(blockCount rpc.DecimalOrHex, lastBlock rpc.BlockNumber, rewardPercentiles []float64) (*types.FeeHistoryResult, error)
```
Comment (ilovers) [inline app/rpc/backend/backend.go]: There is no need to add this method?

File: app/rpc/backend/backend.go [hunk 3]
```diff
@@ -498,3 +501,12 @@ func (b *EthermintBackend) ConvertToBlockNumber(blockNumberOrHash rpctypes.Block
 	}
 	return rpctypes.BlockNumber(out.Number), nil
 }
+
+func (b *EthermintBackend) FeeHistory(
```
Comment (ilovers) [inline app/rpc/backend/backend.go]: There is no need to add this method?

File: app/rpc/namespaces/eth/api.go [hunk 4]
```diff
@@ -1708,3 +1708,58 @@ func (api *PublicEthereumAPI) saveZeroAccount(address common.Address) {
 	zeroAccount.SetBalance(sdk.DefaultBondDenom, sdk.ZeroDec())
 	api.watcherBackend.CommitAccountToRpcDb(zeroAccount)
 }
+
+func (e *PublicEthereumAPI) FeeHistory(blockCount rpc.DecimalOrHex, lastBlock rpc.BlockNumber, rewardPercentiles []float64) (*rpctypes.FeeHistoryResult, error) {
+	e.logger.Debug("eth_feeHistory")
+	return e.backend.FeeHistory(blockCount, lastBlock, rewardPercentiles)
```
Comment (ilovers) [inline app/rpc/namespaces/eth/api.go]:
```suggestion
	return nil, fmt.Errorf("unsupported rpc function: eth_FeeHistory")
```

**Review by KamiD — APPROVED** [review]

**Pull request merged**